modules/core/shared/src/test/scala/ciris/ConfigCodecSpec.scala

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2017-2024 Viktor Rudebeck
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+package ciris
+
+import cats.data.Chain
+import cats.Eq
+import cats.laws.discipline.ContravariantTests
+import cats.laws.discipline.MonadErrorTests
+import cats.syntax.all._
+import org.scalacheck.Arbitrary
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalacheck.Gen
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Test
+import scala.concurrent.duration.{Duration, FiniteDuration}
+import scala.util.Try
+
+final class ConfigCodecSpec extends DisciplineSuite with Generators {
```

Comment:
It would be nice to have roundtrip tests for the `ConfigCodec`s (so `encode` is also tested). Same for the codecs in the integration modules.